qiskit_experiments/library/characterization/zz_ramsey.py

```diff
@@ -0,0 +1,339 @@
+            «                                      0
+
+            Modified Ramsey sequence with qubit 1 initially in the excited state
+
+                 ┌────┐ ░ ┌─────────────────┐ ░ ┌───┐ ░ ┌─────────────────┐ ░ »
+            q_0: ┤ √X ├─░─┤ Delay(delay[s]) ├─░─┤ X ├─░─┤ Delay(delay[s]) ├─░─»
+                 ├───┬┘ ░ └─────────────────┘ ░ ├───┤ ░ └─────────────────┘ ░ »
+            q_1: ┤ X ├──░─────────────────────░─┤ X ├─░─────────────────────░─»
+                 └───┘  ░                     ░ └───┘ ░                     ░ »
+            c: 1/═════════════════════════════════════════════════════════════»
+                                                                              »
+            «     ┌─────────────────────┐┌────┐ ░ ┌─┐
+            «q_0: ┤ Rz(4*delay*dt*f*pi) ├┤ √X ├─░─┤M├
+            «     └─────────────────────┘└────┘ ░ └╥┘
+            «q_1: ──────────────────────────────░──╫─
+            «                                   ░  ║
+            «c: 1/═════════════════════════════════╩═
+            «                                      0
+
+    # section: analysis_ref
+
+        :py:class:`ZZRamseyAnalysis`
+    """
+
+    def __init__(
+        self,
+        qubits: (int, int),
+        backend: Optional[Backend] = None,
+        **experiment_options,
+    ):
+        """Create new experiment.
+
+        Args:
+            qubits: The qubits on which to run the Ramsey XY experiment.
+            backend: Optional, the backend to run the experiment on.
+            experiment_options: experiment options to set
+        """
+        super().__init__(qubits=qubits, analysis=ZZRamseyAnalysis(), backend=backend)
+        # Override the default of get_processor() which is "1" * num_qubits. We
+        # only fit the probability of the target qubit.
+        self.analysis.set_options(outcome="1")
+        self.set_experiment_options(**experiment_options)
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default values for the :math:`ZZ` Ramsey experiment.
+
+        Experiment Options:
+            delays (list[float]): The list of delays that will be scanned in
+                the experiment, in seconds. If not set, then ``num_delays``
+                evenly spaced delays between ``min_delay`` and ``max_delay``
+                are used. If ``delays`` is set, ``max_delay``, ``min_delay``,
+                and ``num_delays`` are ignored.
+            max_delay (float): Maximum delay time to use.
+            min_delay (float): Minimum delay time to use.
+            num_delays (int): Number of circuits to use per control state
+                preparation.
+            zz_rotations (float): The "fake" rotation added to qubit 0 uses a
+                frequency that gives this many rotations in the case where
+                :math:`f_{ZZ}` is 0.
+        """
+        options = super()._default_experiment_options()
+        options.delays = None
+        options.min_delay = 0e-6
```

Comment:
I really like this interface. We should update other experiments too in another PR.